var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -34,7 +34,7 @@ class Mesa(AutotoolsPackage):
     list_url = "https://mesa.freedesktop.org/archive"
     list_depth = 2
 
-    version('17.1.5', '6cf936fbcaadd98924298a7009e8265d')
+    version('17.1.5', '6cf936fbcaadd98924298a7009e8265d', url='https://mesa.freedesktop.org/archive/mesa-17.1.5.tar.xz')
```

Comment:
Why do u need specify url explicitly?

Reply:
I noticed that the URL of mesa version 17.x is different from the previous versions.
We have for 17.x:
https://mesa.freedesktop.org/archive/mesa-17.x.x.tar.xz

For older versions:
https://mesa.freedesktop.org/archive/older-versions/13.x/13.x.x/mesa-13.x.x.tar.xz

Probably it's better to change the list_url and fix the older urls (for 12.x and 13.x versions)

Reply:
I can confirm that `spack fetch mesa` currently fails with `17.1.5`. Probably this change should rather go into 
```
def url_for_version(self, version):
    if version >= Version('13.0.6'):
      ....
    else:
      ....
```


Then you can conditionally use either old one or new one based on the version.

Reply:
ok